src/JwtCredentialsManager.ts

```diff
@@ -19,19 +19,21 @@ export default abstract class JwtCredentialsManager<
 
   token: string | null | undefined;
 
-  credentials: TCredentials | null | undefined;
+  credentialsPromise: Promise<TCredentials | null | undefined> | null;
 
   renewHandle: NodeJS.Timeout | null | undefined;
 
   constructor(config: JwtCredentialsManagerConfig) {
     this.config = config;
 
     this.token = null;
-    this.credentials = null;
+    this.credentialsPromise = null;
   }
 
-  getCredentials(): TCredentials | null | undefined {
-    const { credentials } = this;
+  async getCredentials(): Promise<TCredentials | null | undefined> {
+    if (!this.credentialsPromise) return null;
+
```

Comment:
**suggestion:** not needed, because awaiting `null` should just give you `null`

```suggestion
```

Reply:
Yep I just spotted this one too